torch/csrc/distributed/c10d/default_comm_hooks.cpp

```diff
@@ -19,7 +30,13 @@ c10::intrusive_ptr<c10::ivalue::Future> AllReduceCommHook::runHook(
 c10::intrusive_ptr<c10::ivalue::Future> FP16CompressCommHook::runHook(
     GradBucket& bucket) {
 
-  auto compressed_tensor = bucket.getBufferRef().to(torch::kFloat16);
+  at::Tensor compressed_tensor;
+  if (bucket.getBufferRef().is_complex()) {
+    compressed_tensor = at::view_as_real(bucket.getBufferRef()).to(torch::kFloat16);
+  } else {
+    compressed_tensor = bucket.getBufferRef().to(torch::kFloat16);
```

Comment:
I thought about doing it this way, but it would lead to an extra Tensor temporary in the non-complex case. 

`getBufferRef` returns a `Tensor&` which would be copied to `compressed_tensor`, which is not necessary.